src/Compilers/Core/Portable/DiagnosticAnalyzer/AnalyzerAssemblyLoader.Core.cs

```diff
@@ -227,7 +227,19 @@ internal sealed class CompilerAnalyzerAssemblyResolver(AssemblyLoadContext compi
         {
             private readonly AssemblyLoadContext _compilerAlc = compilerContext;
 
-            public Assembly? ResolveAssembly(AssemblyName assemblyName) => _compilerAlc.LoadFromAssemblyName(assemblyName);
+            public Assembly? ResolveAssembly(AssemblyName assemblyName)
+            {
+                try
+                {
+                    return _compilerAlc.LoadFromAssemblyName(assemblyName);
```

Comment:
I'm wondering what was happening previously, where was the exception propagated (did it crash the whole process)?

Reply:
The code which consumes `IAnalyzerAssemblyResolver` guards against unexpected exceptions. Basically, that is an extension point and the compiler generally guards against unexpected exceptions coming from extensions. So this scenario was worknig overall but this specific component wasn't meeting its contract.